WordPress/Classes/Utility/ReaderPostCacheProvider.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+
+@objc class ReaderPostCacheProvider: NSObject {
+
+    private enum ReaderPostType {
+
+        case normal
+        case cross
+        case attribution
+
+        static let allTypes = [normal, cross, attribution]
+    }
+
+    let context: NSManagedObjectContext
+    init(context: NSManagedObjectContext) {
+        self.context = context
+    }
+
+    /**
+     Gets an stored ReaderPost from the local database, it searches for a Normal Post, 
+     a SourceAttribution Post or a Cross Post.
+     
+     - Parameters:
+        - postID: the postID of the post to fetch
+        - siteID: the siteID of the post to fetch, can be a blogID if it has source attribution information
+     
+     - Returns: A ReaderPost object if it is cached localy or nil if not found
+     */
+    func getPostByID(postID: Int, siteID: Int) -> ReaderPost? {
```

Comment:
Cool, thanks for the clarification.
I will try the predicate block approach with the `isNormalPost` as the last one for sure! 😝